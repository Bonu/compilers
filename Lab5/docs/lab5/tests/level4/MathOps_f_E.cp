// Main: Test expressions of overloaded operators with mixed types.

class Main {

  void main(){

    int i;        int j;
    string a;     string b;
    bool p;       bool q;

  // Assignments 
    i = 2.0;
    a = p;
    q = 1;
    i = a;
    a = p;
    p = 0;

  // Mathematical Expressions 
   // Binary Operators 

    i = a * j;
    i = j + a;		// int and float 
    i = j - 5.0;

    i = j * p;
    i = p / j;		// int and bool 

    p = i;		// bool and int 
    p = b;		// bool and float 

    a = p;

   // Unary Operators 
    p = -p;


  // Relational Expressions 
	p = p > a;
	p = i > q;
	p = p >= q;
	p = (p == a);
  }
}
