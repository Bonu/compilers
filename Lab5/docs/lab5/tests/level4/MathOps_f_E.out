line 12: the target ` i ' and source have incompatible types
          -- expects `int' but found `float'
line 13: the target ` a ' and source have incompatible types
          -- expects `string' but found `bool'
line 14: the target ` q ' and source have incompatible types
          -- expects `bool' but found `int'
line 15: the target ` i ' and source have incompatible types
          -- expects `int' but found `string'
line 16: the target ` a ' and source have incompatible types
          -- expects `string' but found `bool'
line 17: the target ` p ' and source have incompatible types
          -- expects `bool' but found `int'
line 22: invalid argument 1 ` a ' for arithmetic operator `*'
          -- expects a numeric argument but found string
line 23: invalid argument 2 ` a ' for arithmetic operator `+'
          -- expects a numeric argument but found string
line 24: the target ` i ' and source have incompatible types
          -- expects `int' but found `float'
line 26: invalid argument 2 ` p ' for arithmetic operator `*'
          -- expects a numeric argument but found bool
line 27: invalid argument 1 ` p ' for arithmetic operator `/'
          -- expects a numeric argument but found bool
line 29: the target ` p ' and source have incompatible types
          -- expects `bool' but found `int'
line 30: the target ` p ' and source have incompatible types
          -- expects `bool' but found `string'
line 32: the target ` a ' and source have incompatible types
          -- expects `string' but found `bool'
line 35: invalid argument type for unary operator `-'
          -- expects a numeric argument but found bool
line 39: operands to relational operator `>' have incompatible types
line 40: operands to relational operator `>' have incompatible types
line 42: operands to relational operator `==' have incompatible types
18 errors were detected
